portable/GCC/ARM_CRx_MPU/portmacro_asm.h

```diff
@@ -0,0 +1,515 @@
+#define PORTMACRO_ASM_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "FreeRTOSConfig.h"
+
+#ifndef configTOTAL_MPU_REGIONS
+    #error "Set configTOTAL_MPU_REGIONS to the humber of MPU regions in FreeRTOSConfig.h"
+#elif( configTOTAL_MPU_REGIONS == 8 )
+    #define portMPU_TOTAL_REGIONS ( 8UL )
+#elif( configTOTAL_MPU_REGIONS == 12 )
+    #define portMPU_TOTAL_REGIONS ( 12UL )
+#elif( configTOTAL_MPU_REGIONS == 16 )
+    #define portMPU_TOTAL_REGIONS ( 16UL )
+#else
+    #error "Set configTOTAL_MPU_REGIONS to the humber of MPU regions in FreeRTOSConfig.h"
+#endif
+
+/** On the ArmV7-R Architecture the Operating mode of the Processor is set using
+ * the Current Program Status and Control Register (CPSR) Mode bits, [4:0]
+ * the only registers banked between modes are the CPSR, Stack Pointer (R13),
+ * and the Link Register (R14). FIQ mode also banks the GPRs R8-R12
+ * Of note, the only mode not "Privileged" is User Mode
+ *
+ * Additional information about the Processor Modes can be found here:
+ * https://developer.arm.com/documentation/ddi0406/cb/System-Level-Architecture/The-System-Level-Programmers--Model/ARM-processor-modes-and-ARM-core-registers/ARM-processor-modes?lang=en
+ *
+ * */
+
+/** @brief CPSR Mode bit field value for User Mode
+ * @ingroup Port Privilege
+ */
+#define USER_MODE                        0x10U
+
+/** @brief CPSR Mode bit field value for Fast Interrupt Handler (FIQ) Mode
+ * @ingroup Port Privilege
+ */
+#define FIQ_MODE                         0x11U
+
+/** @brief CPSR Mode bit field value for Interrupt Handler (IRQ) Mode
+ * @ingroup Port Privilege
+ */
+#define IRQ_MODE                         0x12U
+
+/** @brief CPSR Mode bit field value for Supervisor (SVC) Mode
+ * @ingroup Port Privilege
+ */
+#define SVC_MODE                         0x13U
+
+/** @brief CPSR Mode bit field value for Monitor (MON) Mode
+ * @ingroup Port Privilege
+ */
+#define MON_MODE                         0x16U
+
+/** @brief CPSR Mode bit field value for Abort (ABT) Mode
+ * @ingroup Port Privilege
+ */
+#define ABT_MODE                         0x17U
+
+/** @brief CPSR Mode bit field value for Hypervisor (HYP) Mode
+ * @ingroup Port Privilege
+ */
+#define HYP_MODE                         0x1AU
+
+/** @brief CPSR Mode bit field value for Undefined (UND) Mode
+ * @ingroup Port Privilege
+ */
+#define UND_MODE                         0x1BU
+
+/** @brief CPSR Mode bit field value for System (SYS) Mode
+ * @ingroup Port Privilege
+ */
+#define SYS_MODE                         0x1FU
+
+/** @brief Mark that a Task Stack has padding values added
+ * @ingroup Task Context
+ */
+#define portSTACK_FRAME_HAS_PADDING_FLAG ( 1UL << 0UL )
+
+/** @brief Flag uses to mark that a FreeRTOS Task is privileged
+ * @ingroup Port Privilege
+ */
+#define portTASK_IS_PRIVILEGED_FLAG      ( 1UL << 1UL )
+
+/** @brief SVC Number to use when requesting a context swap.
+ * @ingroup Scheduler
+ */
+#define portSVC_YIELD                    0x0100
```

Comment:
Its probably worth at least leaving a comment that these values cannot clash with the system call numbers defined in mpu_syscall_numbers.h